src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,47 @@
 package core.basesyntax;
 
 public class SalaryInfo {
-    public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+    public static String getSalaryInfo(String[] names, String[] data,
+                                       String dateFrom, String dateTo) {
+        String dateFromComparable = convertDateToComparable(dateFrom);
+        String dateToComparable = convertDateToComparable(dateTo);
+        int[] totalEarnings = new int[names.length];
+
+        for (String entry : data) {
```

Comment:
Remember about informative names of variables and methods. The variable 'data' is too generic. A name like 'records' or 'workLogs' would be more descriptive.